exec/java-exec/src/main/java/org/apache/drill/exec/planner/DFSDirPartitionLocation.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Class defines a single partition corresponding to a directory in a DFS table.
+ */
+package org.apache.drill.exec.planner;
+
+
+import com.google.common.collect.Lists;
+
+import java.util.Collection;
+import java.util.List;
+
+public class DFSDirPartitionLocation implements PartitionLocation {
+  private final Collection<PartitionLocation> subPartitions;
+  private final String[] dirs;
+
+  public DFSDirPartitionLocation(String[] dirs, Collection<PartitionLocation> subPartitions) {
+    this.subPartitions = subPartitions;
+    this.dirs = dirs;
+  }
+
+  @Override
+  public String getPartitionValue(int index) {
+    assert index < dirs.length;
```

Comment:
this one actually is copied from [1]. I think it makes sense to change both to throw exception in stead of relying on assertion check. Will update the patch. 


https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/planner/DFSPartitionLocation.java#L58